taipy/core/data/data_node.py

```diff
@@ -373,6 +373,28 @@ def _get_last_modified_datetime(cls, path: Optional[str] = None) -> Optional[dat
     def storage_type(cls) -> str:
         raise NotImplementedError
 
+    def _get_downloadable_path(self) -> str:
```

Comment:
this has already defined in the FileDataNodeMixin no?

Reply:
Yes. The idea is that we want the method to only defined in `_ FileDataNodeMixin`.

`SQLDataNode. _get_downloadable_path()` should raise the NotImplementedError error.

Reply:
ah ok, then I would suggest to move this code in to _abstract_sql.py instead, since DataNode and _FileDataNodeMixin are 2 different abstract class, I don't think they will have much connection in this sense.

Reply:
hmm in that case, I think it should still be fine if you put it in the _abstract_sql, since other DataNode are not extending from `_FileDataNodeMixin` so it should be fine for them no?

Reply:
what about other classes like Mongo, S3, Generic, ...?
If we do that, we also need to add one for each of them, which is weird.

Reply:
hmmm I don't think they are extending the class `_FileDataNodeMixin` :/